sdk/tests/conformance2/rendering/framebuffer-mismatched-attachment-targets.html

```diff
@@ -0,0 +1,161 @@
+        gl.bindRenderbuffer(target, rb);
+        gl.renderbufferStorage(target, format, size>>i, size>>i);
+      }
+      ret.attach = (attachEnum, mipLevel) => {
+        const rb = ret.mips[mipLevel];
+        gl.framebufferRenderbuffer(gl.FRAMEBUFFER, attachEnum, gl.RENDERBUFFER, rb);
+      };
+      break;
+    }
+    case gl.TEXTURE_2D:
+    case gl.TEXTURE_CUBE_MAP: {
+      ret = gl.createTexture();
+      gl.bindTexture(target, ret);
+      gl.texStorage2D(target, mipLevels, format, size, size);
+      let imageTarget = target;
+      if (imageTarget == gl.TEXTURE_CUBE_MAP) {
+        imageTarget = gl.TEXTURE_CUBE_MAP_POSITIVE_X+2; // Deliberately don't choose the first image.
+      }
+      ret.attach = (attachEnum, mipLevel) => {
+        gl.framebufferTexture2D(gl.FRAMEBUFFER, attachEnum, imageTarget, ret, mipLevel);
+      };
+      break;
+    }
+    case gl.TEXTURE_3D:
+    case gl.TEXTURE_2D_ARRAY: {
+      ret = gl.createTexture();
+      gl.bindTexture(target, ret);
+      gl.texStorage3D(target, mipLevels, format, size, size, 1);
+      ret.attach = (attachEnum, mipLevel) => {
+        gl.framebufferTextureLayer(gl.FRAMEBUFFER, attachEnum, ret, mipLevel, 0);
+      };
+      break;
+    }
+    default:
+      throw new Error();
+  }
+  ret.target = wtu.glEnumToString(gl, target);
+  ret.format = wtu.glEnumToString(gl, format);
+  return ret;
+}
+
+function runTest() {
+  const MIP_LEVELS = 2;
+  const SIZE = 2;
+
+  gl.clearColor(1, 0, 0, 1);
+
+  const program = wtu.setupProgram(gl, ['vshader','fshader'], [], console.log.bind(console));
+  gl.useProgram(program);
+
+  const colorResList = [
+    newResource(gl.RENDERBUFFER, MIP_LEVELS, gl.RGBA8, SIZE),
+    newResource(gl.TEXTURE_2D, MIP_LEVELS, gl.RGBA8, SIZE),
+    newResource(gl.TEXTURE_CUBE_MAP, MIP_LEVELS, gl.RGBA8, SIZE),
+    newResource(gl.TEXTURE_3D, MIP_LEVELS, gl.RGBA8, SIZE),
+    newResource(gl.TEXTURE_2D_ARRAY, MIP_LEVELS, gl.RGBA8, SIZE),
+  ];
+
+  const depthResList = [
+    newResource(gl.RENDERBUFFER, MIP_LEVELS, gl.DEPTH_COMPONENT16, SIZE),
+    newResource(gl.TEXTURE_2D, MIP_LEVELS, gl.DEPTH_COMPONENT16, SIZE),
+    newResource(gl.TEXTURE_CUBE_MAP, MIP_LEVELS, gl.DEPTH_COMPONENT16, SIZE),
+    //newResource(gl.TEXTURE_3D, MIP_LEVELS, gl.DEPTH_COMPONENT16, SIZE), // Depth formats forbidden for TEXTURE_3D.
+    newResource(gl.TEXTURE_2D_ARRAY, MIP_LEVELS, gl.DEPTH_COMPONENT16, SIZE),
+  ];
+
+  const fb = gl.createFramebuffer();
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
+  for (const color of colorResList) {
+    for (const depth of depthResList) {
+      debug(`\ncolor: ${color.target}; depth: ${depth.target}`);
+      for (let mipLevel = 0; mipLevel < MIP_LEVELS; mipLevel++) {
+        debug(`mipLevel: ${mipLevel}`);
+        color.attach(gl.COLOR_ATTACHMENT0, mipLevel);
+        depth.attach(gl.DEPTH_ATTACHMENT, mipLevel);
+        if (!wtu.framebufferStatusShouldBe(gl, gl.FRAMEBUFFER, [gl.FRAMEBUFFER_COMPLETE, gl.FRAMEBUFFER_UNSUPPORTED])) {
```

Comment:
The cases mentioned earlier still fail on macOS, including in Firefox Nightly. This has to continue here (skip the rendering and clearing tests) if the framebuffer status is FRAMEBUFFER_UNSUPPORTED.

Reply:
It's sort of a pain to spin up a machine to fix this. If you see what's wrong it's most efficient if you can suggest/provide a fix collaboratively!

Reply:
Got it. I'll post a diff here shortly which you can apply.

Reply:
Never mind, your revision is more elegant than the one I was going to propose. :)